Homework/cashier.js

```diff
@@ -0,0 +1,32 @@
+import {Person} from "./person.js";
+import {ShoppingError} from "./shoppingError.js";
+export class Cashier extends Person {
+
+    #totalSales;
+
+    constructor() {
+        super();
+        this.#totalSales = 0;
+    }
+
+    get totalSales() {
+        return this.#totalSales;
+    }
+
+    scanAndCash(customer) {
+        let totalCartPrice = 0;
+        customer.shoppingCart.forEach(product => {totalCartPrice += product.price});
+
+        if (totalCartPrice > customer.money) {
+            throw new ShoppingError({
+                code: ShoppingError.ERROR_CODE.NOT_ENOUGH_MONEY,
+                data: { customer }
+            });
+        }
+
+        customer.money -= totalCartPrice;
```

Comment:
`shoppingCart` and `money` properties should be private and you should only expose them via getters as read-only properties, otherwise any other entity can play with `Customer` internals.

If you want to manipulate them you should implement methods in the `Customer` class and call them where needed.